vmss-prototype/vmss-prototype

```diff
@@ -1028,18 +1259,27 @@ def register_commands(subparser):
     parser = add_command(subparser, "status", func=vmss_prototype_status,
                          description='Show status of VMSS Prototype Pattern')
 
-    parser = add_command(subparser, "update", func=vmss_prototype_update,
-                         description='Set up new/updated prototype')
+    parser = add_update_command(subparser, "update", func=vmss_prototype_update,
+                                description='Set up new/updated prototype')
     parser.add_argument('--target-node', required=True, type=NodeName,
                         help='Target node to use as the prototype source')
-    parser.add_argument('--new-updated-nodes', default=0, type=NewUpdatedNodes,
-                        help='Number of new nodes to add to the cluster after the VMSS has been updated from the prototype node (default: %(default)s)')
-    parser.add_argument('--max-history', default=3, type=HistorySize,
-                        help='Number of entries to keep in history - minimum is 2 - (default: %(default)s)')
-    parser.add_argument('--grace-period', type=int, default=300,
-                        help='Grace period in seconds for drain (default: %(default)s)')
-    parser.add_argument('--force', default=False, action='store_true',
-                        help='Force using target node even if drain fails')
+
+    parser = add_update_command(subparser, "auto-update", func=vmss_prototype_auto_update,
+                                description='Automatically set up new/updated prototype if needed')
+    parser.add_argument('--target-vmss', action='append', nargs='+', type=VmssName,
+                        help='Target VMSS pools (all if not given)')
+    parser.add_argument('--pending-reboot-annotation', required=True, type=str,
+                        help='The name of the annotation that, if it exists, the node is pending reboot')
+    parser.add_argument('--last-patch-annotation', required=True, type=str,
+                        help='The name of the annotation that holds the timestamp at which the last patch was applied')
+    parser.add_argument('--node-skip-annotation', action='append', nargs='+', required=False, type=str,
+                        help='Optional annotations, that if it exists, will disqualify the node as a potential candidate')
+    parser.add_argument('--minimum-ready-time', default=60, type=MinimumReadyTime,
+                        help='Minimum time a node must in ready state before it is considered a candidate (default %(default)s)')
+    parser.add_argument('--minimum-candidates', default=1, type=MinimumCandidates,
+                        help='Minimum number of acceptable candidates (default: %(default)s)')
+    parser.add_argument('--running-on-node', default=os.getenv('NODE_ID', None), type=str,
```

Comment:
How about we just change this to an array of node names, if we ever want to statically "disallow" one or more nodes? The nominal case will be helm auto-populating this with one node.

In which case it might be `--ignore-nodes` (or `--skip-nodes` if we prefer skip over ignore)

Reply:
Interesting - currently it is filled in with the NODE_ID from the environment that gets set up with the yaml.  Will have to look into changing into a multi-value.  Will also have to look how to best deal with them in helm command line (it is not the easiest to deal with lists of things)  Maybe the first step is to just make the command support it and then later deal with helm if we want to.

Reply:
So, what I did was to *always* respect NODE_ID and to make --ignore-nodes to allow multiple node names to be listed (default none) that are ignored